content/3_Silver/Binary_Search.mdx

```diff
@@ -1169,3 +1169,186 @@ for _ in range(queryNum):
+      <Quiz.Explanation>
+        Incorrect. Consider the case where the target element is the median of the array. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      $\mathcal{O}(\log n)$ / $\Omega(1)$
+      <Quiz.Explanation>
+        Correct. The worst case happens when the number is not present, and the best case occurs if the first number we check is the target value. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $\mathcal{O}(n)$ / $\Omega(\log n)$
+      <Quiz.Explanation>
+        Incorrect. Binary search always takes at most $\lceil \log n \rceil$ iterations. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $\mathcal{O}(\log n)$ / $\Omega(\log \log n)$
+      <Quiz.Explanation>
+        Incorrect. Consider the case where the target element is the median of the array. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    What is an advantage of using binary search over linear search?
+    <Quiz.Answer correct>
+      Faster time complexity
+      <Quiz.Explanation>
+        Correct. Linear search runs in $\mathcal{O}(n)$, whereas binary search runs in $\mathcal{O}(\log n)$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Less Memory
+      <Quiz.Explanation>
+        Incorrect. Binary search and linear search both use $\mathcal{O}(n)$ memory.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Easier implementation. 
+      <Quiz.Explanation>
+        Incorrect. Linear search is, in most cases, a simple `for` loop, whereas binary search most often requires at least two additional variables and conditionals. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Suppose a program involving binary search runs in $\mathcal{O}(\log n)$ time for some $n$. What is the smallest value of $n$ where runtime would be an issue? A typical online judge can handle around $10^9$ operations in a second. 
+    <Quiz.Answer>
+      $9$
+      <Quiz.Explanation>
+        Incorrect. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $10^9$
+      <Quiz.Explanation>
+      	Incorrect
+      </Quiz.Explanation>
+    </Quiz.Answer>
+		<Quiz.Answer>
+		$99^9$
+    <Quiz.Explanation>
+    	Incorrect
+    </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      $10^{10^{9}}$
+      <Quiz.Explanation>
+        Correct. The logarithm of this value is a bit more than $10^9$. A pure, logarithmic binary search will almost never run into issues regarding time limit. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Suppose we were tasked to find some maximum value such that a given monotonic boolean function `someBooleanFunction(m)` returns `true`. The following code has an error in it that causes an infinite loop. Find the corrected version. 
```

Comment:
```suggestion
    Suppose we were tasked to find the maximum value `m` in the range `[l,r]` such that a given monotonic boolean function `someBooleanFunction(m)` returns `true`. The following code has an error in it that causes an infinite loop. Find the corrected version. 
```

Reply:
Are we assuming that `0 <= l <= r` and that no overflow ever occurs when we compute `l+r`?

Reply:
what if no such `m` exists?